### PR TITLE
Fix invalid string formatting in merge.py

### PR DESCRIPTION
**Pull request by fvollmer**

### CODE REVIEW

**Review by deeb02 — COMMENTED** [review]
Is this a merely stylistic change? The code AFAICT should work the same before and after the change. What am I missing?

**Comment by gbaned** [comment]
@fvollmer Can you please check @deeb02's comments and keep us posted ? Thanks!

**Comment by fvollmer** [comment]
This is not a stylistic change. The variable `input_shape` is a tuple and for this reason it can cause the error `TypeError: not all arguments converted during string formatting`. Here a simple python example to demonstrate the difference:

```python
my_tuple = (1, 2, 3)
my_string = ("abc %s" % (my_tuple,))  # works fine
my_string = ("abc %s" % my_tuple)  # TypeError: not all arguments converted during string formatting
```

File: tensorflow/python/keras/layers/merge.py [hunk 1]
```diff
@@ -500,7 +500,7 @@ def build(self, input_shape):
     if len(shape_set) != 1:
       err_msg = ('A `Concatenate` layer requires inputs with matching shapes '
                  'except for the concat axis. Got inputs shapes: %s' %
-                 input_shape)
```
[the remark this answers is not in the document]
Comment (fvollmer) [reply tensorflow/python/keras/layers/merge.py:503]: @deeb02 This isn't working fine. I added a comment to illustrate the difference

**Comment by deeb02** [comment]
Thank you for the clarification. Makes sense.

File: tensorflow/python/keras/layers/merge.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (deeb02) [reply tensorflow/python/keras/layers/merge.py:503]: Got it.
Thanks for the fix!

**Review by deeb02 — APPROVED** [review]
Looks good to me.

**Comment by fvollmer** [comment]
What happend here? Seems like my commit f444f868df477470747305debd8bccc757c9e50f was rolled back by the tensorflow gardener 980ed0664ce73bcb1ba4f7edcef3706679a3e482? What was the reason?

**Pull request merged**